src/js/components/AppConfigEditFormComponent.jsx

```diff
@@ -0,0 +1,268 @@
+import React from "react/addons";
+import Util from "../helpers/Util";
+
+import AppsEvents from "../events/AppsEvents";
+import AppFormErrorMessages from "../constants/AppFormErrorMessages";
+import AppFormStore from "../stores/AppFormStore";
+import AppsStore from "../stores/AppsStore";
+import CollapsiblePanelComponent
+  from "../components/CollapsiblePanelComponent";
+import ContainerSettingsComponent
+  from "../components/ContainerSettingsComponent";
+import FormActions from "../actions/FormActions";
+import FormEvents from "../events/FormEvents";
+import HealthChecksComponent from "../components/HealthChecksComponent";
+import OptionalEnvironmentComponent
+  from "../components/OptionalEnviromentComponent";
+import OptionalLabelsComponent from "../components/OptionalLabelsComponent";
+import OptionalSettingsComponent
+  from "../components/OptionalSettingsComponent";
+import FormGroupComponent from "../components/FormGroupComponent";
+
+var AppConfigEditFormComponent = React.createClass({
+  displayName: "AppConfigEditFormComponent",
+
+  propTypes: {
+    app: React.PropTypes.object,
+    onChange: React.PropTypes.func,
+    onError: React.PropTypes.func
+  },
+
+  getDefaultProps: function () {
+    return {
+      app: null,
+      onChange: Util.noop,
+      onError: Util.noop
```

Comment:
I wanted to keep the interface the same for the edit form and the JSON editor.

It could make sense for `onChange` to be required, but `onError` is never called by the JSON editor.